src/components/SpellBox.js

```diff
@@ -0,0 +1,46 @@
+import { Fragment } from "react";
+import { Link, useParams } from "react-router-dom";
+import Spell from "./Spell";
+import Bookmark from "./Bookmark"
+import "../css/SpellBox.css";
+import useFetch from '../hooks/useFetch';
+import endpoints from '../endpoints.js';
+
+const SpellBox = () => {
+  const charClass = useParams().class;
+  const {data, isLoading, error} = useFetch(`https://www.dnd5eapi.co/api/classes/${charClass}/spells`);
+  const charSpells = data?.results;
+
+  let spells;
+  
+  if (!isLoading) {
+    spells = charSpells.map((spell) => {
+      return (
+        <div key={spell.index} className="singleSpell">
+          <Link to={`/spells/${spell.index}`} > 
+              <Spell 
+              name={spell.name}
+              />
+          </Link>
+          <Bookmark />
+        </div>
+      )
+    })
+  }
+
+  return (
+    <div className="spellBox">
+      {error ? <>{error}</> :
+      <>
+        {isLoading ? <>loading...</> :
+          <>
+          {spells}
```

Comment:
Then replace this line here with 
`charSpells.map(spell => <Spell name={spell.name}  /> <Bookmark /> )`

Reply:
We will also need to pass the spell name (or probably index?) to the Bookmark component as well,